Applications/Spire/Include/Spire/Ui/CurrencyComboBox.hpp

```diff
@@ -24,12 +19,14 @@ namespace Spire {
       explicit CurrencyComboBox(const Nexus::CurrencyDatabase& database,
         QWidget* parent = nullptr);
 
-      //! Connects a slot to the Currency selection signal.
-      boost::signals2::connection connect_selected_signal(
-        const SelectedSignal::slot_type& slot) const;
+      //! Returns the CurrencyId of the current Currency.
+      Nexus::CurrencyId get_currency() const;
```

Comment:
The API for UI widgets should as a matter of principle signal when changes are made to its internal data model. So at a minimum there should be signals for when the user has selected a value.